frontend/main.js

```diff
@@ -314,8 +328,25 @@ async function startApp () {
           primaryButton: L('Close')
         })
       }).finally(() => {
-        this.ephemeral.ready = true
-        this.removeLoadingAnimation()
+        // Wait for SW to be ready
+        console.debug('[app] Waiting for SW to be ready')
+        const sw = ((navigator.serviceWorker: any): ServiceWorkerContainer)
+        sw.ready.then(() => {
+          const onready = () => {
```

Comment:
Claude suggests:

> The service worker ready check could be more robust. If the service worker fails to register or activate, this promise might never resolve. Consider adding a timeout:


```js
Promise.race([
  sw.ready,
  new Promise((_, reject) => setTimeout(() => reject(new Error('SW ready timeout')), 10000))
]).then(() => {
  // ... existing code ...
}).catch(e => {
  console.error('Service worker failed to become ready:', e)
  // Fallback behavior
  this.ephemeral.ready = true
  this.removeLoadingAnimation()
})
```

Reply:
You could add an `alert()` with a message explaining the situation. This way if someone encounters this they can tell us and we'll know it's a thing